nitro-protocol/contracts/VirtualPaymentApp.sol

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.7.6;
+pragma experimental ABIEncoderV2;
+
+import './interfaces/IForceMoveApp.sol';
+import './libraries/NitroUtils.sol';
+import './interfaces/INitroTypes.sol';
+import {ExitFormat as Outcome} from '@statechannels/exit-format/contracts/ExitFormat.sol';
+
+/**
+ * @dev The VirtualPaymentApp contract complies with the ForceMoveApp interface and allows payments to be made virtually from Alice to Bob (participants[0] to participants[1]).
+ */
+contract VirtualPaymentApp is IForceMoveApp {
+    struct VoucherAmountAndSignature {
+        uint256 amount;
+        INitroTypes.Signature signature; // signature on abi.encode(channelId,amount)
+    }
+
+    enum AllocationIndices {
+        Alice, // payer
+        Bob // beneficiary, initial allocation is zero
+    }
+
+    /**
+     * @notice Encodes application-specific rules for a particular ForceMove-compliant state channel.
+     * @dev Encodes application-specific rules for a particular ForceMove-compliant state channel.
+     * @param fixedPart Fixed part of the state channel.
+     * @param proof Array of recovered variable parts which constitutes a support proof for the candidate.
+     * @param candidate Recovered variable part the proof was supplied for.
+     */
+    function requireStateSupported(
+        FixedPart calldata fixedPart,
+        RecoveredVariablePart[] calldata proof,
+        RecoveredVariablePart calldata candidate
+    ) external pure override {
+        // This channel has only 4 states which can be supported:
+        // 0 prefund
+        // 1 postfund
+        // 2 redemption
+        // 3 final
+
+        // states 0,1,3 can be supported via unanimous consensus:
+
+        if (proof.length == 0) {
+            require(
+                NitroUtils.getClaimedSignersNum(candidate.signedBy) ==
+                    fixedPart.participants.length,
+                '!unanimous; |proof|=0'
```

Comment:
Should we make these `require` strings easier to parse? Or is this a notation we use all over the place?

Reply:
The reason they are a bit difficult to parse is because I am trying to keep them short to save gas. I need to refresh my memory about whether the gas costs are only significant for deployment, though.

Probably we should be adopting a more modern approach to errors https://docs.soliditylang.org/en/latest/contracts.html#errors-and-the-revert-statement .